.github/workflows/gobuild.yml

```diff
@@ -0,0 +1,99 @@
+name: build
```

Comment:
I quite like this change, but am not convinced about the rest (with the exception of time.Since). Mind going through the PR again and checking what is worth keeping when merging from your fork?